src/remote/addr/mod.rs

```diff
@@ -91,10 +95,17 @@ impl RemoteAddr {
         unimplemented!("So far, it is not possible to use this method!")
     }
 
-    pub fn send<T: RemoteMessage + Serialize>(&self, _msg: T) {
-        unimplemented!(
-            "So far, it is not possible to receive responses from remote destinations as futures!"
-        )
+    pub fn send<T, R, H>(&self, msg: T, c: &Addr<H>) -> Receiver<Box<dyn Any + Send>>
+    where
+        T: RemoteMessage + Serialize,
+        R: RemoteMessage + Send + 'static,
+        H: Handler<R> + Handler<ResponseSubscribe>,
+        <H as Actor>::Context: ToEnvelope<H, ResponseSubscribe>,
+    {
+        let (tx, rx) = oneshot::channel();
+        c.do_send(ResponseSubscribe(TypeId::of::<R>(), tx));
+        self.do_send(msg);
```

Comment:
How does the receiver know who to answer?

Reply:
Actor, to which address `c` is pointing guaranteed to handle remote `R` message. There is no guarantee that remote system would send some response or send response to this particular actor though, so it's up to developer.
I think we could upgrade this conception further and provide more guarantees